src/index.js

```diff
@@ -154,6 +155,110 @@ VirtualMachine.prototype.loadProject = function (json) {
     this.runtime.setEditingTarget(this.editingTarget);
 };
 
+VirtualMachine.prototype.toPrettyJSON = function (testing) {
+    this.runtime.targets.testing = testing;
+    var json = JSON.stringify(this.toJSON(), null, 4);
+    this.runtime.targets.testing = false;
+    if (testing == true) {
+        console.log('Exported To JSON');
+    }
+    return json;
+};
+
+VirtualMachine.prototype.toJSON = function () {
+    if (this.runtime.targets.hasOwnProperty('testing') == false) {
+        this.runtime.targets.testing = false;
+    }
+    var obj = new Object();
+    obj.sprites = this.runtime.targets;
+    obj.meta = new Object();
+    obj.meta.name = 'WIP';
+    obj.meta.useragent = navigator.userAgent;
+    return obj;
+};
+
+VirtualMachine.prototype.fromJSON = function (json, testing) {
+    this.clear();
+    if (testing == true) {
+        console.log('Importing JSON');
+    }
+    var obj = JSON.parse(json);
+    var i = 0;
+    for (; i < obj.sprites.length; i++) {
+        var blocks = new Blocks();
+        var z = null;
+        for (z in obj.sprites[i].sprite.blocks) {
+            blocks[z] = obj.sprites[i].sprite.blocks[z];
+            if (testing == true) {
+                console.log('Importing ' + z + ' to Blocks');
+            }
+        }
+        var sprite = new Sprite(blocks, this.runtime);
+        var y = null;
+        for (y in obj.sprites[i].sprite) {
+            if (y === 'blocks') {
+                continue;
+            }
+            sprite[y] = obj.sprites[i].sprite[y];
+            if (testing == true) {
+                console.log('Importing ' + y + ' to Sprite');
+            }
+        }
+        var target = sprite.createClone();
+        this.runtime.targets.push(target);
+        var x = null;
+        for (x in obj.sprites[i]) {
+            if (x === 'sprite') {
+                continue;
+            }
+            target[x] = obj.sprites[i][x];
+            console.log('Importing ' + x + ' to Rendered Target');
+        }
+        target.updateAllDrawablePropertie();
+    }
+    this.editingTarget = this.runtime.targets[0];
+    // Update the VM user's knowledge of targets and blocks on the workspace.
+    this.emitTargetsUpdate();
+    this.emitWorkspaceUpdate();
+    this.runtime.setEditingTarget(this.editingTarget);
+    if (testing == true) {
+        console.log('Imported from JSON');
+    }
+};
+
+VirtualMachine.prototype.testJSON = function (debug) {
```

Comment:
When @rschamp was requesting tests he was referring to automated unit tests. This should be removed from the VM interface.